osu.Game.Tests/NonVisual/Filtering/FilterMatchingTest.cs

```diff
@@ -78,6 +78,20 @@ public void TestCriteriaMatchingConvertedBeatmaps()
             Assert.IsFalse(carouselItem.Filtered.Value);
         }
 
+        [Test]
+        public void TestCriteriaMatchingConvertedBeatmapsForCustomRulesets()
+        {
+            var exampleBeatmapInfo = getExampleBeatmap();
+            var criteria = new FilterCriteria
+            {
+                Ruleset = new RulesetInfo { OnlineID = -25 },
```

Comment:
```suggestion
                Ruleset = new RulesetInfo { OnlineID = -1 },
```

No ruleset should be using anything but -1 (the automatic default) so the test should also specify this.